app/scripts/modules/core/src/pipeline/config/stages/bakeManifest/ManualExecutionBakeManifest.tsx

```diff
@@ -29,7 +31,7 @@ export function ManualExecutionBakeManifest(props: ITriggerTemplateComponentProp
   React.useEffect(() => {
     if (overrideArtifact === false) {
       removeHelmArtifact();
-    } else {
+    } else if (defaultArtifact !== null) {
```

Comment:
Do we need to also add `defaultArtifact` as a dependency of the `useEffect`?  (Or is that only for things that are part of the react state whereas this is a local variable?)

Reply:
This question made me start asking 5️⃣ ❓ about this component and ended up on a possibly regrettable journey; I'm going to push another commit that cleans it up a bit and will let you know when this is ready for another look.

(But yes, I think your instinct was right because the general advice is that for effects with explicitly specified dependencies, they should be exhaustive)

Reply:
Okay, pushed a cleanup commit for this manual execution component (and tested with both legacy/standard artifacts UI). Still think the overall experience of supplying artifacts to override is pretty confusing but am drawing the line here so this PR doesn't get out of control.